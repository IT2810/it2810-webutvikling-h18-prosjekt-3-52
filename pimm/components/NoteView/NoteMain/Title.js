import React from 'react';
import { Text, TextInput, View, Button, AsyncStorage } from 'react-native';
import SaveButton from '../NoteTopBar/SaveButton.js';

export default class Title extends React.Component{
	constructor(props){
		super(props);
		this.state={
			title: "",
		};
		
	}
	
	changeTitle(newTitle){
		this.setState({title: newTitle});
	
	}
	
	handleChange(newTitle){
		this.changeTitle(newTitle);
		this.props.sendTitleUp(newTitle);
	}
	
	componentWillReceiveProps(nextProps){
		if(this.props.title!==nextProps.title || nextProps.fl){
			this.setState({title: nextProps.title});
			this.props.sfl();
			
		}
	}
	
	
	
	render() {
		return(
			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', padding: 5 }}>
			<TextInput 
			placeholder="Write title here"
			style={{padding: 10, width: 280, backgroundColor: 'beige', fontWeight: 'bold', fontSize: 20}}
			onChangeText={(title) => this.handleChange(title)}
			value={this.state.title}
			/>
			</View>		
		);
	}
	
}