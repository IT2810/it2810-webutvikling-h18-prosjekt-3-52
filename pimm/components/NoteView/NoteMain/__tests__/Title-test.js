import React from 'react';
import 'react-native';
import Title from '../Title.js';
import renderer from 'react-test-renderer';


test('title renders correctly', () => {
  const titleRendered = renderer.create(<Title />).toJSON();
  expect(titleRendered).toMatchSnapshot();
  expect(titleRendered.type).toBe("View");
  expect(titleRendered.children[0].type).toBe("TextInput");
  expect(titleRendered.children[0].props.placeholder).toBe("Write title here");
  
});

test('Change title', ()=>{
	const title = renderer.create(<Title />).getInstance();
	expect(title.state.title).toBe("");
	title.changeTitle("Original");
	expect(title.state.title).toBe("Original");
});
