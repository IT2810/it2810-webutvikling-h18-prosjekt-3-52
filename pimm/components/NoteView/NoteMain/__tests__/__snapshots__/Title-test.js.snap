// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Write title here"
    style={
      Object {
        "backgroundColor": "beige",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 10,
        "width": 280,
      }
    }
  />
</View>
`;
